src/ValidatorRule.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atk4\Validate;
+
+use Atk4\Data\Exception;
+use Atk4\Data\Model;
+
+class ValidatorRule
```

Comment:
This is probably not needed to align /w atk4, thus it should be not part of this PR.

the correct way would be:
- one PR to fix CS/stan (for atk4/ui 4.0)
- do 4.0 release
- then PR to align /w atk4/ui 5.0
- if refactoring is wanted, one or more isolated PRs
- test manually extensively as I doubt everything is tested using CI in this repo
- do 5.0 release

Reply:
This lib allow bridging an easy with valitron.
we need Atk4/Data in __construct to link Validate hook and in validate method we have Atk4\Data\Model as first argument.
Actually use very simple methods:
`$model->onHook`
`$model->get()`
`$model->get($fieldName)`

i create a PR that fix CS but still have a problem with stan : #13 

But probably due to simple use of Model we can allow dependency of Atk4/Data > 3.1.0 but in this way we need to test against 3.1 4.0 5.0 or thinking this is a marginal repo, do only validation and use really solid and low level methods of Model, we can probably leave as it is WDYT?

Reply:
if (upcoming) rel 4.0 will pass all tests /w atk4/data 3.1, we can allow both (3.1 + 4.0)